tests/Unit/CallPowerTest.php

```diff
@@ -0,0 +1,184 @@
+            'number_dialed_into' => '+12028519273',
+        ]);
+
+        $response->assertSuccessful();
+
+        Bus::assertDispatched(CreateCallPowerPostInRogue::class);
+    }
+
+    /**
+     * Test that a post with a busy status successfully sends to Rogue.
+     *
+     * @return void
+     */
+    public function testBusyCallStatus()
+    {
+        // Fake the job being dispatched.
+        Bus::fake();
+
+        // Create a post with completed as the status.
+        $response = $this->withChompyApiKey()->json('POST', 'api/v1/callpower/call', [
+            'mobile' => '1234567891',
+            'callpower_campaign_id' => '1',
+            'status' => 'busy',
+            'call_timestamp' => '2017-11-09 06:34:01.185035',
+            'call_duration' => 50,
+            'campaign_target_name' => 'Mickey Mouse',
+            'campaign_target_title' => 'Representative',
+            'campaign_target_district' => 'FL-7',
+            'callpower_campaign_name' => 'Test',
+            'number_dialed_into' => '+12028519273',
+        ]);
+
+        $response->assertSuccessful();
+
+        Bus::assertDispatched(CreateCallPowerPostInRogue::class);
+    }
+
+    /**
+     * Test that a post with an invalid CallPower campaign id is not made.
+     *
+     * @return void
+     */
+    public function testFailedPostWithInvalidCallPowerCampaignId()
+    {
+        // Mock the Northstar call.
+        $this->mockGetUserNSCall();
+
+        // If the provided Action ID doesn't exist in Rogue, this method throws:
+        $this->rogueMock->shouldReceive('getActionIdFromCallPowerCampaignId')->andThrow(new Exception(500));
+
+        // Since we don't have an action in Rogue, we shouldn't make a post:
+        $this->rogueMock->shouldNotReceive('createPost');
+
+        $this->expectException(Exception::class);
+
+        CreateCallPowerPostInRogue::dispatch([
+            'mobile' => '1234567891',
+            'callpower_campaign_id' => 1,
+            'status' => 'busy',
+            'call_timestamp' => '2017-11-09 06:34:01.185035',
+            'call_duration' => 50,
+            'campaign_target_name' => 'Mickey Mouse',
+            'campaign_target_title' => 'Representative',
+            'campaign_target_district' => 'FL-7',
+            'callpower_campaign_name' => 'Test',
+            'number_dialed_into' => '+12028519273',
+        ]);
+    }
+
+    /**
+     * Test a failed record if the post is not able to be created.
+     *
+     * @return void
+     */
+    public function testFailedPostCreationInRogue()
+    {
+        // Mock the Northstar call.
+        $this->mockGetUserNSCall();
+
+        // Mock the call to Rogue to get the action id.
+        $this->rogueMock->shouldReceive('getActionIdFromCallPowerCampaignId');
+
+        // Mock a failed call to Rogue when a post is not created.
+        $this->rogueMock->shouldReceive('createPost')->andThrow(new Exception(500));
+
+        $this->expectException(Exception::class);
```

Comment:
This test doesn't effectively do all that much (since it just asserts that we got the exception we mocked above), so it maybe can be removed. The other "failure" tests seem more helpful, since they're asserting that "if this step fails, it shouldn't try to do the other thing".